model_fitting/random_forest.py

```diff
@@ -18,145 +17,357 @@
 
 
 def parse_data(file_path):
-    # reading the CSV file if it's legal
     try:
         data = pd.read_csv(file_path, engine='python')
     except Exception as e:
         exit(f'Cannot analyse data! {e}')
+    
+    # create masks for separating train and test samples (fifth sample of each mAb)    
+    test_rows_mask = data['sample_name'].str.contains('test')
+    train_rows_mask = ~test_rows_mask
 
     # separating train and test samples (fifth sample of each mAb)
-    train_data = data[~data['sample_name'].str.contains('test')]
-    test_data = data[data['sample_name'].str.contains('test')]
-
-    # set sample names as index
-    train_data.set_index('sample_name', inplace=True)
-    test_data.set_index('sample_name', inplace=True)
-
-    return train_data, test_data
-
+    # a matrix of the actual feature values
+    y_train = data['label'][train_rows_mask]
+    y_test = data['label'][test_rows_mask]
+
+    sample_names_train = data['sample_name'][train_rows_mask]
+    sample_names_test = data['sample_name'][test_rows_mask]
+
+    # don't need the labels and sample names anymore
+    data.drop(['sample_name', 'label'], axis=1, inplace=True)
+    # a matrix of the actual feature values
+    X_train = data[train_rows_mask].values
+    X_test = data[test_rows_mask].values
+
+    feature_names = np.array(data.columns)
+
+    return X_train, y_train, X_test, y_test, feature_names, sample_names_train, sample_names_test
+
+
+def get_hyperparameters_grid(seed):
+    # Number of trees in random forest
+    n_estimators = [int(x) for x in np.linspace(start=100, stop=2000, num=20)]
```

Comment:
Set parameters using command arguments, parameters shouldn't be hardcoded.
This remark should be applied to entire file, not just this line